src/maker_protocol.rs

```diff
@@ -553,12 +553,20 @@ fn handle_sign_receivers_contract_tx(
     let mut sigs = Vec::<Signature>::new();
     for receivers_contract_tx_info in message.txes {
         sigs.push(
-            wallet
+            if let Some(c) = wallet
                 .read()
                 .unwrap()
-                .find_swapcoin(&receivers_contract_tx_info.multisig_redeemscript)
-                .ok_or(Error::Protocol("multisig_redeemscript not found"))?
-                .sign_contract_tx_with_my_privkey(&receivers_contract_tx_info.contract_tx)?,
+                .find_incoming_swapcoin(&receivers_contract_tx_info.multisig_redeemscript)
+            {
+                c.sign_contract_tx_with_my_privkey(&receivers_contract_tx_info.contract_tx)?
+            } else {
+                wallet
+                    .read()
+                    .unwrap()
+                    .find_outgoing_swapcoin(&receivers_contract_tx_info.multisig_redeemscript)
+                    .ok_or(Error::Protocol("multisig_redeemscript not found"))?
+                    .sign_contract_tx_with_my_privkey(&receivers_contract_tx_info.contract_tx)?
+            },
```

Comment:
Searching both kinds of swapcoins is wrong here. You should only search the list containing OutgoingSwapCoins (I think), because they are the only ones relevant for the SignReceiversContractTx.